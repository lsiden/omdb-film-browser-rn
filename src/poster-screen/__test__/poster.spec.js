import React from "react"
import renderer from "react-native-test-utils"

import { posterScreen as PosterScreen } from "../"

const defaultProps = () => {
  return {
    navigation: {
      state: {
        uri: "https://my.url.org"
      },
      navigate: () => {}
    }
  }
}

const createWrapper = (props = {}) =>
  renderer(<PosterScreen {...{ ...defaultProps(), ...props }} />)

it("renders", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
