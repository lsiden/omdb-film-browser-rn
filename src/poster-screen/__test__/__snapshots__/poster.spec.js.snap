// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipe={undefined}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    loadingIndicatorSource={
      Object {
        "testUri": "../../../src/assets/spinningwheel-300x216.gif",
      }
    }
    resizeMode="contain"
    source={
      Object {
        "uri": "https://my.url.org",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <View
    hintWrapperStyle={
      Object {
        "alignContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "darkgray",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Dismiss by swiping.
    </Text>
  </View>
</View>
`;
